sklearn/tree/_tree.pyx

```diff
@@ -438,7 +438,7 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
         n_node_samples = end - start
         is_leaf = ((depth > self.max_depth) or
                    (n_node_samples < self.min_samples_split) or
-                   (n_node_samples < 2 * self.min_samples_leaf) or
+                   (n_node_samples < self.min_samples_leaf) or
```

Comment:
I feel a way to fix this now is to check if the weights are uniform and appropriately set `min_samples_split` to be double `min_samples_leaf` or `min_samples_weight`.. But that seems messy